reactor-core/src/main/kotlin/reactor/core/publisher/MonoFunctions.kt

```diff
@@ -17,36 +17,17 @@
 package reactor.core.publisher
 
 import org.reactivestreams.Publisher
-import reactor.util.function.*
 
-/**
- * Aggregates this [Iterable] of void [Publisher]s into a new [Mono].
- * An alias for a corresponding [Mono.when] to avoid use of `when`, which is a keyword in Kotlin.
- *
- * @author DoHyung Kim
- * @since 3.1
- */
-fun Iterable<Publisher<Void>>.whenComplete(): Mono<Void> = Mono.`when`(this)
-
-/**
- * Merges this [Iterable] of [Mono]s into a new [Mono] by combining them
- * with [combinator].
- *
- * @author DoHyung Kim
- * @since 3.1
- */
-@Suppress("UNCHECKED_CAST")
-inline fun <T, R> Iterable<Mono<T>>.zip(crossinline combinator: (List<T>) -> R): Mono<R> =
-        Mono.zip(this) { combinator(it.asList() as List<T>) }
 
 /**
  * Aggregates the given void [Publisher]s into a new void [Mono].
  * An alias for a corresponding [Mono.when] to avoid use of `when`, which is a keyword in Kotlin.
  *
  * @author DoHyung Kim
+ * @author Sebastien Deleuze
  * @since 3.1
  */
-fun whenComplete(vararg sources: Publisher<Void>): Mono<Void> = MonoBridges.`when`(sources)
+fun whenComplete(vararg sources: Publisher<*>): Mono<Void> = MonoBridges.`when`(sources)
```

Comment:
doesn't it break the public binary API? Maybe we should add an overloaded method instead?

Reply:
I have asked on Kotlin Slack to be sure.

Reply:
Generic type will be erased so this is not a breaking change.